Meadowlark-js/backends/meadowlark-opensearch-backend/src/repository/UpdateOpensearch.ts

```diff
@@ -22,6 +22,33 @@ const moduleName = 'opensearch.repository.UpdateOpensearch';
  */
 type OpensearchRequest = { index: string; id: string };
 
+type MapperParsingException = {
+  error: {
+    root_cause: [{ type: string; reason: string }];
+    type: string;
+    reason: string;
+    caused_by: [{ type: string; reason: string }];
+  };
+};
+
+// Exported for the sake of unit testing
+export function safelyLogOpenSearchErrors(err: MapperParsingException | object): object {
+  const mapperParsingException = err as MapperParsingException;
+  if (mapperParsingException != null) {
```

Comment:
Not sure I understand how this function is dealing with types. If `mapperParsingException` can be null, that can only be because `err` can be null, which isn't in the parameter type signature. Note that line 36 is `const mapperParsingException = err` at runtime, as Typescript type information is stripped out.

If it's not guaranteed that the shape of the thrown exception will match the MapperParsingException, you'll have to check that each property you want to use is not undefined.

Separately, if you know that the thrown object is at least an `Error`, that would be preferable to type `object`.

Reply:
Reverse order: `Error` is easy enough in the signature.

At some point I suspected I was getting confused between TypeScript and C# again with the casting. Got distracted and forgot about the uncertainty.

I was looking for the easy way of "if the object contains this keyA, and object[keyA] contains keyB" etc.

Reply:
I'm going to remove this code entirely, since we don't have a clear path for testing it. There is a backlog ticket for making OpenSearch more robust. We need to seriously investigate that, and need to figure out how to force some errors so that we can understand the structure. Looking at the source code from opensearch-js, for example, I don't even see a `root_cause` is defined as a string in TypeScript - but I saw an error message that had it as an array. So there are weird things going on. I'll make a note in that other ticket that we need to watch for potential PII in error messages.